cloudinit/net/network_state.py

```diff
@@ -1052,4 +1020,31 @@ def mask_and_ipv4_to_bcast_addr(mask, ip):
     return bcast_str
 
 
+def parse_net_config_data(net_config, skip_broken=True) -> NetworkState:
+    """Parses the config, returns NetworkState object
+
+    :param net_config: curtin network config dict
+    """
+    state = None
+    version = net_config.get('version')
+    config = net_config.get('config')
+    if version == 2:
+        # v2 does not have explicit 'config' key so we
+        # pass the whole net-config as-is
+        config = net_config
+
+    if version and config is not None:
+        nsi = NetworkStateInterpreter(version=version, config=config)
+        nsi.parse_config(skip_broken=skip_broken)
+        state = nsi.get_network_state()
+
+    if not state:
+        raise RuntimeError(
```

Comment:
Previously, we were either returning `NetworkState` or `None`. In `net_convert.py` we handled the `None` accordingly, but the renderers don't check for `None` and we'll likely get an ugly `AttributeError` error if they're passed it. So failing early here makes more sense and makes the function easier to reason about.